src/runtime_src/core/edge/drm/zocl/sched_exec.c

```diff
@@ -2208,6 +2198,28 @@ fini_scheduler_thread(void)
 	return retval;
 }
 
+static inline int
+dma_done(struct sched_cmd *cmd)
+{
+	if (cmd->dma_handle.dma_flags & ZOCL_DMA_DONE) {
+		cmd->dma_handle.dma_flags &= ~ZOCL_DMA_DONE;
```

Comment:
Any particular reason we have to clear DONE bit here? I think DONE and ERROR flags should have same life cycle and you can clear them altogether in your penguin query call.